packages/contracts/contracts/rewards/RewardsManager.sol

```diff
@@ -349,29 +360,34 @@ contract RewardsManager is RewardsManagerV4Storage, GraphUpgradeable, IRewardsMa
      * @return Assigned rewards amount
      */
     function takeRewards(address _allocationID) external override returns (uint256) {
-        // Only Staking contract is authorized as caller
-        IStaking staking = staking();
-        require(msg.sender == address(staking), "Caller must be the staking contract");
+        address rewardsIssuer = msg.sender;
+        require(rewardsIssuers[rewardsIssuer], "Caller must be a rewards issuer");
+
+        (
+            address indexer,
+            bytes32 subgraphDeploymentID,
+            uint256 tokens,
+            uint256 accRewardsPerAllocatedToken
+        ) = IRewardsIssuer(rewardsIssuer).getAllocationData(_allocationID);
 
-        IStaking.Allocation memory alloc = staking.getAllocation(_allocationID);
-        uint256 accRewardsPerAllocatedToken = onSubgraphAllocationUpdate(alloc.subgraphDeploymentID);
+        uint256 updatedAccRewardsPerAllocatedToken = onSubgraphAllocationUpdate(subgraphDeploymentID);
```

Comment:
This will call `getAccRewardsPerAllocatedToken` which is still getting the amount of allocated tokens from staking only. I think `getAccRewardsPerAllocatedToken` should be updated to also fetch from the Subgraph Service and add them?

Reply:
fixed here: https://github.com/graphprotocol/contracts/pull/982